src/modules/dashboard/components/RecoveryModeDialog/RecoveryModeDialogForm.tsx

```diff
@@ -0,0 +1,158 @@
+    defaultMessage: 'Enter Recovery mode',
+  },
+  recoveryModeDescription: {
+    id:
+      'admin.RecoveryModeDialog.RecoveryModeDialogForm.recoveryModeDescription',
+    defaultMessage: `If you believe that something dangerous is happening in
+    your colony (e.g. it is under attack), recovery mode will disable the colony
+    and prevent further activity until the issue has been overcome.`,
+  },
+  leavingRecoveryModeDescription: {
+    id:
+      // eslint-disable-next-line max-len
+      'admin.RecoveryModeDialog.RecoveryModeDialogForm.leavingRecoveryModeDescription',
+    defaultMessage: `
+    Leaving recovery requires the approval of a majority of members
+    holding the {roleRequired} permission. <a>Learn more.</a>`,
+  },
+  annotation: {
+    id: 'dashboard.RecoveryModeDialog.RecoveryModeDialogForm.annotation',
+    defaultMessage:
+      'Explain why you’re putting this colony into recovery mode (optional)',
+  },
+  noPermission: {
+    id: 'dashboard.RecoveryModeDialog.RecoveryModeDialogForm.noPermission',
+    defaultMessage: `You do not have the {roleRequired} permission required
+      to take this action.`,
+  },
+});
+
+interface Props {
+  back: () => void;
+  colony: Colony;
+}
+
+const RecoveryModeDialogForm = ({
+  back,
+  colony,
+  handleSubmit,
+  isSubmitting,
+}: Props & FormikProps<FormValues>) => {
+  const { walletAddress, username, ethereal } = useLoggedInUser();
+
+  const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
+
+  const hasRegisteredProfile = !!username && !ethereal;
+
+  const userHasPermission =
+    hasRegisteredProfile && canEnterRecoveryMode(allUserRoles);
+
+  return (
+    <>
+      <DialogSection appearance={{ theme: 'heading' }}>
+        <Heading
+          appearance={{ size: 'medium', margin: 'none' }}
+          text={MSG.title}
+          className={styles.title}
+        />
+      </DialogSection>
+      {!userHasPermission && (
+        <DialogSection>
+          <PermissionRequiredInfo requiredRoles={[ColonyRole.Recovery]} />
+        </DialogSection>
+      )}
+      <DialogSection>
+        <FormattedMessage {...MSG.recoveryModeDescription} />
+      </DialogSection>
+      <DialogSection>
+        <div className={styles.permissionLabel}>
+          <FormattedMessage
+            {...MSG.leavingRecoveryModeDescription}
+            values={{
+              roleRequired: (
+                <PermissionsLabel
+                  permission={ColonyRole.Recovery}
+                  name={{
+                    id: `role.${ColonyRole.Recovery}`,
+                  }}
+                />
+              ),
+              a: (chunks) => (
+                <Link href="/" to="/" target="_blank" className={styles.link}>
+                  {chunks}
+                </Link>
+              ),
+            }}
+          />
+        </div>
+      </DialogSection>
+      <DialogSection>
+        <Annotations
+          label={MSG.annotation}
+          name="annotations"
```

Comment:
This is a minor nitpick. It's singular. `annotation` vs. `annotations`

But it doesn't matter that much, so I'm fine with this either way :)